auth/internal/transport/cba.go

```diff
@@ -151,7 +169,18 @@ func GetHTTPTransportConfig(opts *Options) (cert.Provider, func(context.Context,
 		return nil, nil, err
 	}
 
-	if config.s2aAddress == "" {
+	var s2aAddr string
+	var transportCredsForS2A credentials.TransportCredentials
+
+	if config.mtlsS2AAddress != "" {
+		s2aAddr = config.mtlsS2AAddress
+		transportCredsForS2A, err = loadMTLSMDSTransportCreds(mtlsMDSRoot, mtlsMDSKey)
+		if err != nil {
```

Comment:
1. Why is this `err` thrown away?
2. Why is the default (non-s2a) configuration being returned in case of an error from `loadMTLSMDSTransportCreds`, rather than falling through to set up `config.s2aAddress`?

Reply:
1. thanks for catching this, added a log line

2. in each environment GKE/GCE/Serverless, we either serve mtls_s2a_address or plaintext_s2a_address, not both (there's only one s2a gRPC server running in the infrastructure) 
so if `config.mtlsS2AAddress` is provided, there's no need to try the other `config.s2aAddress`.

Reply:
Got it, that makes sense.